src/main/java/org/apache/commons/csv/CSVRecord.java

```diff
@@ -203,18 +238,48 @@ public boolean isSet(final String name) {
      * @return an iterator over the values of this record.
      */
     @Override
-    public Iterator<String> iterator() {
+    public final Iterator<String> iterator() {
         return toList().iterator();
     }
+    
+    /**
+     * Return a mutable CSVRecord.
+     * <p>
+     * Mutable records have more efficient implementations of 
+     * {@link #withValue(int, String)} and {@link #withValue(String, String)}
+     * for when multiple modifications are to be done on the same record.
+     * <p>
+     * If this record is already mutable, it will be returned directly, otherwise
+     * a copy of its values will be made for the new mutable record. 
+     * <p>
+     * Use {@link #immutable()} to freeze a mutable CSVRecord. 
+     * 
+     * @return A mutable CSVRecord
+     */
+    public final CSVRecord mutable() {
+    	if (isMutable()) {
+    		return this;
+    	}
+		String[] newValues = Arrays.copyOf(values, values.length);
+    	return new CSVMutableRecord(newValues, mapping, comment, recordNumber, characterPosition);
+	}    
+
+    final void put(final int index, String value) {
+        values[index] = value;
```

Comment:
Should it be possible to modify columns outside the length of `values`?  (Would cause multiple array resizes if done many times)